proposals/TP3-xds-error-propagation.md

```diff
@@ -0,0 +1,122 @@
+
+The xDS Management server is only expected to return the error message once rather than throughout for future responses. The client is expected to remember the error message until either a new error message is returned or the resource is returned. 
+
+### Wildcard Resources
+
+It is possible to subscribe to all resources by a client using a wildcard or "" resource name. The control plane in this case can provide error details for two different use cases. One when the issue is with the glob itself or later on when the issue is specific to individual resources. 
+
+1. For errors associated with wildcard("" for legacy or "*") and for xDS-TP glob collections; the control plane `error_details` resource name will match the relevant wildcard request("" or "*" or xDS-TP glob collections). This can be used by the control plane to indicate an error with the collection as a whole.
+2. For errors associated with specific individual resources that match the glob,
+    * The resource name should be the specific resource name associated with the error  OR
+    * The control could just not use this mechanism for wildcard subscriptions, because if the client doesn't have permission to access a resource, then it probably shouldn't be considered to match the wildcard subscription to begin with.
+
+## Rationale
+
+The major alternative to this proposal is to use Wrapped Resources by using Resource Containers defined in https://www.envoyproxy.io/docs/envoy/latest/xds/core/v3/resource.proto#xds-core-v3-resource. 
+
+### Wrapped Resources
+
+xDS resource containers are the default protos used for the Incremental xDS protocol and it's usage in SoTW is controlled via the client feature `xds.config.supports-resource-in-sotw`. 
+
+In this proposal the error information is directly passed as part of the resource field in the Resource Container, using the artifact that the field is a protobuf.Any. This enables us to designate the resource as either a `ResourceError` if the xDS management server encountered problems, or as the actual `Resource` if no errors occurred. 
+
+#### Backward Compatibility
+
+To avoid possible confusion with this behavior, it must be protected with a client feature similar to `supports-resource-in-sotw` called `supports-resource-error-unwrapping`. 
+
+Note: This should also be documented here: https://www.envoyproxy.io/docs/envoy/latest/api/client_features#currently-defined-client-features
+
+#### Why not this approach?
+
+This approach has two major drawbacks compared to the chosen approach:
+
+* This alternative would not work for non-wrapped resources
+* It introduces a backwards compatibility issue, which adding a new field wouldn’t have as clients would just ignore them. 
+
+## Implementation
+
+This will probably be implemented in gRPC before Envoy.
+
+## Open issues (if applicable)
+
+### Possible improvements to NACKs
```

Comment:
I think this issue has nothing to do with this proposal, since it's talking about a very different problem.  I suggest just removing this.

Reply:
Done